### PR TITLE
osd/OSD: only wake up empty pqueue.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by majianpeng** [comment]
retest this please

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -10513,6 +10513,7 @@ void OSD::ShardedOpWQ::_enqueue(OpQueueItem&& item) {
   sdata->shard_lock.lock();
 
   dout(20) << __func__ << " " << item << dendl;
+  bool empty = sdata->pqueue->empty();
```
Comment (xiexingguo) [inline src/osd/OSD.cc]: this looks wrong to me.
we might still have to process the on_commit callbacks even if the op_wq is empty.

**Comment by xiexingguo** [comment]
I think the primary benefit is that this can save us a redundant and expensive notify_one system call, since in most cases we'll try to keep the op_wq busy (non-empty).

**Comment by majianpeng** [comment]
@neha-ojha . update  by your suggestions.

**Comment by majianpeng** [comment]
@neha-ojha . ping.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-07-22_16:29:12-rados-wip-kefu-testing-2019-07-22-1948-distro-basic-mira/

http://pulpito.ceph.com/kchai-2019-07-24_01:28:15-rados-wip-kefu-testing-2019-07-22-1948-distro-basic-smithi/

- "evicting unresponsive client": tracked by http://tracker.ceph.com/issues/40867
- tasks.mgr.test_progress.TestProgress: addressed by #29111
- tasks.mgr.dashboard.test_health.HealthTest: already fixed by #29172

**Pull request merged**